src/Application.cc

```diff
@@ -842,10 +842,19 @@ std::vector<std::pair<std::string, std::vector<std::string>>>
     {
       auto plugin = common::basename(*dirIter);
 
-      // All we verify is that the file starts with "lib", any further
-      // checks would require loading the plugin.
+      // All we verify is that the file starts with shared library prefix and
+      // ends with shared library suffix, any further checks would require
+      // loading the plugin.
 
-      if (plugin.find("lib") == 0)
+      // TODO(anyone): Move this logic into gz-plugin to be reusable
+
+      // This computation could underflow the unsigned range, but that is okay
+      // as in such case we would check if the suffix is placed somewhere much
+      // further than allowed filename length.
+      const auto suffixPos = plugin.length() - strlen(SHARED_LIBRARY_SUFFIX);
+
+      if (plugin.find(SHARED_LIBRARY_PREFIX) == 0 &&
+          plugin.find(SHARED_LIBRARY_SUFFIX) == suffixPos)
```

Comment:
I do not expect any plugin to have `.so` or `.dll`, but perhaps as it is easy it may be slightly more robust:

```suggestion
          plugin.rfind(SHARED_LIBRARY_SUFFIX) == suffixPos)
```

Reply:
Oh, I wanted to use it, it somehow slipped! Thanks.